alacritty/src/event.rs

```diff
@@ -554,14 +554,23 @@ impl<N: Notify + OnResize> Processor<N> {
                     },
                     WindowEvent::KeyboardInput { input, is_synthetic: false, .. } => {
                         processor.key_input(input);
-                        if input.state == ElementState::Pressed {
-                            // Hide cursor while typing
-                            if processor.ctx.config.ui_config.mouse.hide_when_typing {
-                                processor.ctx.window.set_mouse_visible(false);
-                            }
+                        // Hide cursor while typing, when triggering a keybind that writes to pty
+                        if input.state == ElementState::Pressed
+                            && processor.ctx.config.ui_config.mouse.hide_when_typing
+                            && *processor.ctx.received_count() > 0
+                        {
+                            processor.ctx.window.set_mouse_visible(false);
+                        }
+                    },
+                    WindowEvent::ReceivedCharacter(c) => {
+                        processor.received_char(c);
+                        // Hide cursor while typing
+                        if processor.ctx.config.ui_config.mouse.hide_when_typing
+                            && !*processor.ctx.suppress_chars()
+                        {
+                            processor.ctx.window.set_mouse_visible(false);
                         }
```

Comment:
Why do we do this here instead of the `received_char` method? Seems like this would just duplicate the `suppress_chars` check, which seems inefficient.